src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -106,12 +106,23 @@ public abstract class UntypedObjectBase : IUntypedObject
         /// <summary>
         /// Set a property on the object.
         /// </summary>
-        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyName">Property identifier.</param>
         /// <param name="value">FormulaValue to be set.</param>
         public virtual void SetProperty(string propertyName, FormulaValue value)
         {
             // In case of unwanted behavior, throw an CustomFunctionErrorException exception.
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        /// Set a property on the object.
+        /// </summary>
+        /// <param name="index">Property identifier.</param>
+        /// <param name="value">FormulaValue to be set.</param>
+        public virtual void SetProperty(int index, FormulaValue value)

```

Comment:
>SetProperty [](http://example.com/codeflow?start=28&length=11)

Just to avoid confusion, we should call this a different name.  like:
SetIndex() 

And have a test explicitly that 
Set(uo.'1') calls SetProperty and not gets smart and calls SetIndex.